.travis.yml

```diff
@@ -1,41 +1,160 @@
               # platform started cloning the intended commit."
 
+# NOTE: {os} x {jobs} - {exclude} = {build matrix}
+os:
+  - linux
+  - osx
+
+#  2020-05-26: NOTE: Currently newest Travis dist Ubuntu 18.04 bionic
+dist: bionic
+
+# #  2020-05-26: NOTE: Currently newest macOS image
+# osx_image: xcode11.4  # NOTE: Official Nix installer fails spectacularly on it.
+
 env:
   global:
-    - ALL_TESTS=yes
+    # NOTE: This is secure CACHIX_SIGNING_KEY=val
     - secure: "dm6I+M4+V+C7QMTpcSADdKPE633SvmToXZrTbZ7miNDGmMN+/SfHeN2ybi1+PW6oViMlbPN/7J/aEfiGjSJI8vLk72Y4uCWGmpSb8TXZLu6+whnxtZzzW8+z4tsM4048QJg7CF3N/25U8thRFgs3DqUub1Sf3nG9LrNWdz6ZcDQ="
+    # NOTE: Nix by default uses nixpkgs-unstable channel
+    # NOTE: Setup for Nixpkgs revision
+    # `rev` vals in order of freshness -> cache & stability (branch causes differ):
+    # { master
+    # , commitHash
+    # , haskell-updates
+    # , nixpkgs-unstable
+    # , nixos-unstable
+    # , nixos-20.03
+    # }
+    # - rev=nixos-unstable
+    # # NOTE: Switching into Nixpkgs revision
+    # - NIX_PATH="nixpkgs=https://github.com/nixos/nixpkgs/archive/$rev.tar.gz"
+    # NOTE: Project/binary name
+    - name=hnix
+    # NOTE: Used in the `generateOptparseApplicativeCompletions = true`
+    - pkgName='haskellPackages.hnix'
+    # NOTE: Enable all our tests in cabal
+    - ALL_TESTS=yes
+    # NOTE: Turn all warn into err with {-Wall,-Werror}
+    - failOnAllWarnings=false
+    # NOTE: checkUnusedPackages: is `failOnAllWarnings` + `cabal sdist` to ensure all needed files are listed in the Cabal file. Uses `packunused` or GHC internals. Adds a post-build check to verify that dependencies declared in the cabal file are actually used. The first attrset argument can be used to configure the strictness of this check and a list of ignored package names that would otherwise cause false alarms.
+    - checkUnusedPackages=false
+    # NOTE: Generation and installation of a coverage report.
+    # See https://wiki.haskell.org/Haskell_program_coverage
+    - doCoverage=false
+    # NOTE: Generation and installation of haddock API documentation
+    - doHaddock=false
+    # NOTE: Escape the version bounds from the cabal file. You may want to avoid this function.
+    - doJailbreak=false
+    # NOTE: Disables Nix dependency checking, compilation and execution of test suites listed in the package description file.
+    - doCheck=true
+    # NOTE: doBenchmark: Dependency checking + compilation and execution for benchmarks listed in the package description file.
+    - doBenchmark=false
+    - enableExecutableProfiling=false
+    - enableLibraryProfiling=false
+    # NOTE: Build a source distribution tarball instead of using the source files directly. The effect is that the package is built as if it were published on hackage. This can be used as a test for the source distribution, assuming the build fails when packaging mistakes are in the cabal file.
+    - buildFromSdist=false
```

Comment:
Actually, I'd intuitively say that `true` might be the better default here. Why did you go with `false`?

Reply:
Well, that is just too obvious.

If you put psychologically yourself in my place, even as in Haskell PR terms, you would start to understand.

In short - my main goal is to bring the good CI, hat means - it must be merged to be brought.

Reply:
Sorry, I have no idea. It probably doesn't matter that much, and it certainly shouldn't further block this PR. :)